pkgs/development/compilers/llvm/default.nix

```diff
@@ -53,25 +58,29 @@ let
     in
     lib.nameValuePair attrName (
       recurseIntoAttrs (
-        callPackage ./common {
-          inherit (stdenvAdapters) overrideCC;
-          buildLlvmTools = buildPackages."llvmPackages_${attrName}".tools;
-          targetLlvmLibraries =
-            targetPackages."llvmPackages_${attrName}".libraries or llvmPackages."${attrName}".libraries;
-          targetLlvm = targetPackages."llvmPackages_${attrName}".llvm or llvmPackages."${attrName}".llvm;
-          stdenv =
-            if (lib.versions.major release_version == "13" && stdenv.cc.cc.isGNU or false) then
-              gcc12Stdenv
-            else
-              stdenv; # does not build with gcc13
-          inherit bootBintoolsNoLibc bootBintools;
-          inherit
-            officialRelease
-            gitRelease
-            monorepoSrc
-            version
-            ;
-        }
+        callPackage ./common (
+          {
+            inherit (stdenvAdapters) overrideCC;
+            buildLlvmTools = buildPackages."llvmPackages_${attrName}".tools;
+            targetLlvmLibraries =
+              targetPackages."llvmPackages_${attrName}".libraries or llvmPackages."${attrName}".libraries;
+            targetLlvm = targetPackages."llvmPackages_${attrName}".llvm or llvmPackages."${attrName}".llvm;
+            inherit
+              officialRelease
+              gitRelease
+              monorepoSrc
+              version
+              ;
+          }
+          // packageSetArgs # Allow overrides.
```

Comment:
should this go after the stdenv so it can get overridden too?

Reply:
My thinking was that this involves conditional logic to pull in a different stdenv; that conditional logic still needs to apply. If the user wants to override `gcc12Stdenv` or `stdenv` they still can, through the original argument mechanism. What we don't want is to override the stdenv which has been conditionally set to gcc12Stdenv with the original stdenv.

Reply:
>  What we don't want is to override the stdenv which has been conditionally set to gcc12Stdenv with the original stdenv.

i think that is exactly what we want -- an override that does an override. anyway, i don't think this needs needs to hold up merging given it is pretty minor and the llvm which pins gcc12 will eventually get dropped.

Reply:
FWIW i don't think this comment should've been marked as `resolved` but i am unable to change to `unresolved`.

Reply:
Marked unresolved for you. I think we need someone to break the tie. My point is that the recipe has conditional logic which pulls in the correct stdenv required to build a given LLVM version. If I splat the args in after the conditional logic, the original stdenv would override the conditional logic. This is clearly not what we want. So, I reason that if we put the conditional logic after the args, it's still possible for someone to apply overrides; they just need to override gcc12Stdenv on the input if the conditional logic is in effect.

Reply:
ok. i am just being dense. moving packageSetArgs below the conditional breaks llvm13 as clobbers stdenv . so that won't work.

tho -- i am getting build failures just trying to build llvmPackages_13 from this PR with no modifications.

Reply:
Thanks for testing - can you confirm that the commit on which this is based will build llvmPackages_13? (If not just say and I'll do it when I get a chance)

Reply:
but maybe it was marked broken on master.
```
error: Package ‘lldb-13.0.1’ in  is marked as broken, refusing to evaluate.

```

Reply:
> Thanks for testing - can you confirm that the commit on which this is based will build llvmPackages_13? (If not just say and I'll do it when I get a chance)

4f807e8940284ad7925ebd0a0993d2a1791acb2f appears to have the same issue
```
error: build of '/nix/store/4813qnnz6ymba266v0yjas8b8j6faggl-llef-1.2.0.drv', '/nix/store/s3n5dm0s2x57yr18mqrzy8r0j1nvhzi3-lldb-13.0.1.drv' failed
```